### PR TITLE
[ISSUE-240] Implement metrics for volumemgr.Discovery

### PR DESCRIPTION
**Pull request by ykulazhenkov**
## Purpose
#240

Implement metrics collection for node > volumemgr.Discovery

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

Testing:
```
root@baremetal-csi-node-hl2fd:/# curl -s http://127.0.0.1:8787/metrics  2>&1 | grep discovery
# HELP discovery_drive_count last drive count discovered
# TYPE discovery_drive_count gauge
discovery_drive_count 5
# HELP discovery_duration_seconds duration of the discovery method for the drive manager
# TYPE discovery_duration_seconds histogram
discovery_duration_seconds_bucket{le="0.005"} 0
discovery_duration_seconds_bucket{le="0.01"} 0
discovery_duration_seconds_bucket{le="0.025"} 0
discovery_duration_seconds_bucket{le="0.05"} 16
discovery_duration_seconds_bucket{le="0.1"} 33
discovery_duration_seconds_bucket{le="0.25"} 43
discovery_duration_seconds_bucket{le="0.5"} 43
discovery_duration_seconds_bucket{le="1"} 44
discovery_duration_seconds_bucket{le="2.5"} 44
discovery_duration_seconds_bucket{le="5"} 44
discovery_duration_seconds_bucket{le="10"} 44
discovery_duration_seconds_bucket{le="+Inf"} 45
discovery_duration_seconds_sum 112.37385219999997
discovery_duration_seconds_count 45
```

```
...
reconcile_duration_seconds_bucket{type="node_drive_controller",le="45"} 5
reconcile_duration_seconds_bucket{type="node_drive_controller",le="90"} 5
reconcile_duration_seconds_bucket{type="node_drive_controller",le="180"} 5
reconcile_duration_seconds_bucket{type="node_drive_controller",le="+Inf"} 5
reconcile_duration_seconds_sum{type="node_drive_controller"} 1.069513454
reconcile_duration_seconds_count{type="node_drive_controller"} 5
reconcile_duration_seconds_bucket{type="node_volume_controller",le="0.025"} 5
reconcile_duration_seconds_bucket{type="node_volume_controller",le="0.05"} 5
reconcile_duration_seconds_bucket{type="node_volume_controller",le="0.1"} 5
reconcile_duration_seconds_bucket{type="node_volume_controller",le="0.5"} 5
...
```

### CODE REVIEW

**Comment by ykulazhenkov** [comment]
This PR and https://github.com/dell/csi-baremetal/pull/242 should use same approach for metrics collection

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -26,30 +26,43 @@ import (
 // Statistic is a common interface for histogram metrics
 type Statistic interface {
 	Collect() prometheus.Collector
-	EvaluateDuration(method string) func()
+	EvaluateDuration(labels prometheus.Labels) func()
+	EvaluateDurationForMethod(method string) func()
```
Comment (mishoyama) [inline pkg/metrics/metrics.go:30]: These method names look a little confusing to me. Can we rename?
```
EvaluateDurationForMethod -> EvaluateMethodDuration
EvaluateDurationForType -> EvaluateMethodDurationWithType
```

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -26,30 +26,43 @@ import (
...
+	EvaluateDurationForType(t string) func()
 }
 
+// ExtendedDefBuckets is a default buckets used for csi driver
+var ExtendedDefBuckets = []float64{.025, .05, .1, .5, 1, 3, 5, 10, 15, 30, 45, 90, 180}
```
Comment (mishoyama) [inline pkg/metrics/metrics.go:35]: Should this bucket be metrics specific?

**Review by mishoyama — APPROVED** [review]

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (ykulazhenkov) [reply pkg/metrics/metrics.go:30]: These function are not intent to measure method execution duration. They can be used for these purpose also, but this is only one of possible use scenarios. These functions are simple helpers, they do not Evaluate Method Duration. 

EvaluateDurationForMethod -> helps to evaluate duration of something and labeling it with label "method". EvaluateDurationAndAddLableMethod is the most precise name, but it is obviously too long and ugly.

File: pkg/metrics/metrics.go (hunk 2 above)
Comment (ykulazhenkov) [reply pkg/metrics/metrics.go:35]: Yes, buckets settings can be specific for metric. This is just a default.

**Pull request merged**